content/courses/Ada_For_The_C_Embedded_Developer/chapters/04_Embedded.rst

```diff
@@ -561,6 +561,459 @@ atomic type. Objects of this type |mdash| such as :ada:`R` in this example
+  type;
+
+- we use a representation clause to specify that the :ada:`PMC_SCER` register
+  is at byte #0 and the :ada:`PMC_SCDR` register is at byte #2.
+
+  - Since these registers have 16 bits, we use a range of bits from 0 to 15.
+
+The actual power management controller becomes accessible by the declaration of
+the :ada:`PMC_Periph` object of :ada:`PMC_Peripheral` type. Here, we specify
+the actual address of the memory-mapped registers (`400E0600` in hexadecimal)
+using the :ada:`Address` aspect in the declaration. When we use the
+:ada:`Address` aspect in an object declaration, we're indicating the address in
+memory of that object.
+
+Because we specify the address of the memory-mapped registers in the
+declaration of :ada:`PMC_Periph`, this object is now an overlay for those
+registers. This also means that any operation on this object corresponds to an
+actual operation on the registers of the power management controller. We'll
+discuss more details about overlays in the section about
+:ref:`mapping structures to bit-fields <Mapping_Structures_To_Bit_Fields>` (in
+chapter 6).
+
+Finally, in a test application, we can access any bit of any register of the
+power management controller with simple record component selection. For
+example, we can set the :ada:`USBCLK` bit of the :ada:`PMC_SCER` register by
+using :ada:`PMC_Periph.PMC_SCER.USBCLK`:
+
+.. code:: ada project=Courses.Ada_For_C_Embedded_Dev.Embedded.Register_Overlay_1
+
+    with Registers;
+
+    procedure Set_Bit_In_Register is
+    begin
+       Registers.PMC_Periph.PMC_SCER.USBCLK := 1;
+    end Set_Bit_In_Register;
+
+This code example makes use of many aspects and keywords of the Ada language.
+One of them is the :ada:`Volatile` aspect, which we've discussed in the section
+about :ref:`volatile and atomic objects <VolatileAtomicData>`. Using the
+:ada:`Volatile` aspect for the :ada:`PMC_SCER_Register` type ensures that
+objects of this type won't be stored in a register.
+
+In the declaration of the :ada:`PMC_SCER_Register` record type of the example,
+we use the :ada:`Bit_Order` aspect to specify the bit ordering |mdash| also
+known as `endianess <https://en.wikipedia.org/wiki/Endianness>`_ |mdash| of the
+record type. Here, we can select one of these options:
+
+- :ada:`High_Order_First` (big endian): first bit of the record is the most
+  significant bit;
+
+- :ada:`Low_Order_First` (little endian): first bit of the record is the least
+  significant bit.
+
+The declarations from the :ada:`Registers` package also makes use of aspects
+that we haven't seen yet. Aspects :ada:`Size` and :ada:`Import` will be
+discussed in the section that explains how to
+:ref:`map structures to bit-fields <Mapping_Structures_To_Bit_Fields>` in
+chapter 6. Please refer to that chapter for more details. This is a brief
+explanation of these aspects:
+
+- The :ada:`Size` aspect indicates the number of bits required to represent an
```

Comment:
I think that the first part of your explanation is quite interesting for the readers because it clarifies some details of the source-code example. Therefore, I've just added it to the section (as an admonition).